Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -9171,6 +9172,10 @@ void ByteCodeParser::parse()
     parseCodeBlock();
     linkBlocks(inlineStackEntry.m_unlinkedBlocks, inlineStackEntry.m_blockLinkingTargets);
 
+    m_graph.determineReachability();
+    performLiveCatchVariablePreservationPhase(m_graph);
+    m_graph.clearReachability();
+
     // We run backwards propagation now because the soundness of that phase
     // relies on seeing the graph as if it were an IR over bytecode, since
     // the spec-correctness of that phase relies on seeing all bytecode uses.
```

Comment:
Literally, this is the reason why we need to insert catch preservation before backward propagation.